atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -179,10 +179,11 @@ private static void eagerlyCleanupReadBuffersFromIdleConnection(CassandraClient
         }
     }
 
-    private static boolean isInvalidClientConnection(Exception ex) {
+    private static boolean isInvalidClientConnection(Throwable ex) {
```

Comment:
This is still fragile. What if someone catches an exception and doesn't rethrow? Or throws without rewrapping? Or what if the exception ended up in suppressed exceptions because of a try-with-resources block? Please see my first comment on the issue for a proposal of a more robust fix. Essentially, we want to catch as close to the thrift call as possible.

Reply:
Agree a more robust fix is better and should be implemented.  But is there value in shipping this now to fix most of the problems while we implement the more difficult fix?